envisage/plugins/ipython_kernel/internal_ipkernel.py

```diff
@@ -77,15 +78,37 @@ class InternalIPKernel(HasStrictTraits):
     #: This is a list of tuples (name, value).
     initial_namespace = List()
 
-    def init_ipkernel(self, gui_backend):
+    def init_ipkernel(self, gui_backend=None):
         """ Initialize the IPython kernel.
 
         Parameters
         ----------
-        gui_backend -- string
-          The GUI mode used to initialize the GUI mode. For options, see
-          the `ipython --gui` help pages.
+        gui_backend -- string, optional
+            The GUI mode used to initialize the GUI event loop integration. For
+            options, see the `ipython --gui` help pages. If not given, no event
+            loop integration is set up.
+
+        .. note:: Use of this argument is deprecated!
         """
+        # For backwards compatibility, we allow a kernel to be initialised
+        # twice, and we ignore the second initialization, but warn.
+        if self.ipkernel is not None:
+            warnings.warn(
+                "IPython kernel is being initialised for a second time. "
+                "This may become an error in future versions of this package.",
```

Comment:
Do you also want to mention/instead mention that the kernel has already been initialized and that does nothing?

It's not exactly clear from "IPython kernel is being initialised for a second time." that the `init_ipkernel` method doesn't do anything when called a second/nth time.

Reply:
Yes, I'll update the message to clarify.

Reply:
Done.